clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -981,26 +982,28 @@ private void onFailure(final long currentTimeMs,
                                final Errors responseError) {
             log.debug("Offset fetch failed: {}", responseError.message());
             onFailedAttempt(currentTimeMs);
```

Comment:
It seems `onFailedAttempt` should be called only once by one RPC. However, `OffsetCommitRequestState` could call `onFailedAttempt` several times when the partition-level error is `TOPIC_AUTHORIZATION_FAILED`. 

https://github.com/apache/kafka/blob/c30615e6d7732b0214d59985ee5961bc9266b364/clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java#L725

@lianetm WDYT?

Reply:
you're right, that's definitely a bug that will mess up the logic for retries/backoff in cases of multiple errors at the partition level Filing a jira now...